File: backend/src/main/java/hu/szakdoga/backend/forum/data/dto/MessageDTO.java
package hu.szakdoga.backend.forum.data.dto;

import hu.szakdoga.backend.authentication.data.model.UserEntity;
import hu.szakdoga.backend.forum.data.MessageTypeEnum;

import java.util.Date;

public class MessageDTO {
    public final Long id;
    public final boolean pinned;
    public final String content;
    public final Date dateOfUpload;
    public final MessageTypeEnum type;
    public final Long userId;
    public final Long forumId;

    public MessageDTO(Long id, boolean pinned, String content, Date dateOfUpload, MessageTypeEnum type, Long userId, Long forumId) {
        this.id = id;
        this.pinned = pinned;
        this.content = content;
        this.dateOfUpload = dateOfUpload;
        this.type = type;
        this.userId = userId;
        this.forumId = forumId;
    }
}

File: frontend/src/app/shared/model/forum/message.dto.ts
import { MessageTypeEnum } from './message-type.enum';

export class MessageDto {
    public readonly id: number | null;
    public readonly pinned: boolean;
    public readonly content: string;
    public readonly dateOfUpload: Date;
    public readonly type: MessageTypeEnum;
    public readonly userId: number;
    public readonly forumId: number;

    constructor(
        pinned: boolean,
        content: string,
        type: MessageTypeEnum,
        dateOfUpload: Date,
        userId: number,
        forumId: number,
        id: number | null = null
    ) {
        this.id = id;
        this.pinned = pinned;
        this.userId = userId;
        this.content = content;
        this.type = type;
        this.forumId = forumId;
        this.dateOfUpload = dateOfUpload;
    }
}
